src/main/java/com/dnastack/wes/files/RunFileService.java

```diff
@@ -0,0 +1,242 @@
+            path = evaluateJsonPath(runId, path);
+        }
+
+        final String finalPath = path;
+        RunFile file = getRunFiles(runId).runFiles()
+            .stream()
+            .filter(runFile -> Objects.equals(runFile.getPath(), finalPath))
+            .findFirst()
+            .orElseThrow(() -> new NotFoundException("File %s was not an input or output of the run %s and is not accessible".formatted(finalPath, runId)));
+
+        BlobMetadata metadata = storageClient.getBlobMetadata(file.getPath());
+        file.setBlobMetadata(metadata);
+        return file;
+    }
+
+    private String evaluateJsonPath(String runId, String jsonPath) {
+        String metadataResponse;
+        try {
+            metadataResponse = OBJECT_MAPPER.writeValueAsString(cromwellService.getRun(runId));
+        } catch (JsonProcessingException e) {
+            throw new NotFoundException("Could not find value corresponding to json path expression %s".formatted(jsonPath));
+        }
+        JsonNode foundNode = JsonPath.using(jsonPathConfiguration).parse(metadataResponse).read(jsonPath);
+        if (foundNode != null && foundNode.isTextual()) {
+            return foundNode.asText();
+        } else {
+            throw new NotFoundException("Could not find value corresponding to json path expression %s".formatted(jsonPath));
+        }
+    }
+
+    public void streamRunFile(HttpServletResponse response, String runId, String path, HttpRange httpRange) throws IOException {
+        RunFile runFile = getRunFile(runId, path);
+        long contentLength = runFile.getBlobMetadata().getSize();
+        if (httpRange != null) {
+            long start = httpRange.getRangeStart(runFile.getBlobMetadata().getSize());
+            long end = httpRange.getRangeEnd(runFile.getBlobMetadata().getSize());
+            contentLength = end - start + 1;
+            if (contentLength != runFile.getBlobMetadata().getSize()) {
+                response.setStatus(HttpServletResponse.SC_PARTIAL_CONTENT);
+            }
+        }
+        response.setContentType(MediaType.APPLICATION_OCTET_STREAM_VALUE);
+        response.setHeader("Content-Length", String.valueOf(contentLength));
+        if (runFile.getBlobMetadata().getLastModifiedTime() != null) {
+            response.setHeader("Last-Modified", String.valueOf(runFile.getBlobMetadata().getLastModifiedTime().toEpochMilli()));
+        }
+        storageClient.readBytes(response.getOutputStream(), runFile.getPath(), httpRange);
+    }
+
+    private void extractSecondaryLogFiles(Set<String> secondaryFileSet, Set<String> logFileSet, CromwellMetadataResponse metadataResponse) {
+        Map<String, Object> outputs = metadataResponse.getOutputs();
+        if (outputs != null && !outputs.isEmpty()) {
+            outputs.values().forEach(output -> extractFilesFromValue(secondaryFileSet, OBJECT_MAPPER.valueToTree(output)));
+        }
+        Map<String, List<CromwellTaskCall>> calls = metadataResponse.getCalls();
```

Comment:
Should this be hardcoded to Cromwell?

Reply:
The WES service only deals with cromwell at this point so think its okay